### PR TITLE
Reduce risk of fetching checks in a loop

### PR DESCRIPTION
**Pull request by mtsgrd**
- weird store/state behavior observed in dev that caused loop
- introduced checks reduce risk of runaway requests
- added global rate-limiter for outbound github api requests

---
This is **part of a stack** made with GitButler:
- #5568 👈 
- #5561

### CODE REVIEW

**Comment by mtsgrd** [comment]
@ndom91 any chance you could have a look at this one?

**Comment by ndom91** [comment]
> @ndom91 any chance you could have a look at this one?

Yeah I'll take a look first thing tmr 👍

File: apps/desktop/src/lib/branch/SeriesHeader.svelte [hunk 1]
```diff
@@ -98,11 +97,18 @@
 		sourceBranch && shouldCheck ? $forge?.checksMonitor(sourceBranch) : undefined
 	);
 
-	// Trigger refresh of pull request status and checks when branch(es) are pushed.
+	// Extra reference to avoid potential infinite loop.
+	let lastSeenPush: Date | undefined;
+
+	// Without lastSeenPush this code has code into an infinite loop, where lastPush seemingly
```
Comment (ndom91) [inline apps/desktop/src/lib/branch/SeriesHeader.svelte]: `This code has code into an infinite loop`? :joy:

**Review by ndom91 — APPROVED** [review]
Another tiny typo in `ratelimit.ts`, I think you meant `timestamps` not `timestampts`, right?

Otherwise LGTM :+1:

File: apps/desktop/src/lib/branch/SeriesHeader.svelte (hunk 1 above)
Comment (mtsgrd) [reply apps/desktop/src/lib/branch/SeriesHeader.svelte]: I'm glad you read the comments! 🤗

**Comment by mtsgrd** [comment]
Thank you!

**Pull request merged**